fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationSettings.kt

```diff
@@ -0,0 +1,148 @@
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+            ),
+    ) {
+        LazyColumn {
+            items(translationSwitchList) { item: TranslationSwitchItem ->
+                SwitchWithLabel(
+                    checked = item.isEnabled,
+                    onCheckedChange = item.onStateChange,
+                    label = item.textLabel,
+                    modifier = Modifier
+                        .padding(start = 72.dp, end = 16.dp)
+                        .clearAndSetSemantics {
+                            contentDescription = item.contentDescription
+                        },
+                )
+
+                if (item.hasDivider) {
+                    Divider(Modifier.padding(top = 8.dp, bottom = 8.dp))
+                }
+            }
+        }
+
+        Text(
+            text = stringResource(
+                id = R.string.translation_settings_fragment_translation_preference,
+            ),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 72.dp, end = 16.dp, bottom = 16.dp, top = 8.dp),
+            color = FirefoxTheme.colors.textAccent,
+            style = FirefoxTheme.typography.headline8,
+        )
+
+        TextListItem(
+            label = stringResource(id = R.string.translation_settings_fragment_automatic_translation),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 56.dp)
+                .clickable { onAutomaticTranslationClicked() },
+        )
+
+        TextListItem(
+            label = stringResource(
+                id = R.string.translation_settings_fragment_automatic_never_translate_site,
+            ),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 56.dp)
+                .clickable { onNeverTranslationClicked() },
+        )
+
+        TextListItem(
+            label = stringResource(
+                id = R.string.translation_settings_fragment_download_language,
+            ),
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 56.dp)
+                .clickable { onDownloadLanguageClicked() },
+        )
+    }
+}
+
+/**
+ * Return a list of Translation option switch list item.
+ */
+@Composable
+fun getTranslationSettingsSwitchList(): List<TranslationSwitchItem> {
+    return mutableListOf<TranslationSwitchItem>().apply {
+        add(
+            TranslationSwitchItem(
+                isEnabled = true,
+                onStateChange = {},
+                hasDivider = false,
+                contentDescription = getSwitchContentDescription(
+                    textSwitchId = R.string.translation_settings_fragment_offer_to_translate_content_description,
+                    isEnabled = true,
+                ),
+                textLabel = stringResource(R.string.translation_settings_fragment_offer_to_translate),
```

Comment:
Could we reorder the params here to align with how we have defined the param order in `TranslationSwitchItem `

Reply:
Done